catbird-api/src/main/java/com/redmadrobot/catbird/api/Catbird.java

```diff
@@ -0,0 +1,114 @@
+package com.redmadrobot.catbird.api;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+
+import com.google.gson.Gson;
+
+public final class Catbird {
+    private final URL url;
+
+    public Catbird(final URL url) {
+        this.url = url;
+    }
+
+    public URL url() {
+        return this.url;
+    }
+
+    public void update(final CatbirdMock mock) {
+        RequestPattern pattern = mock.pattern();
+        ResponseMock response = mock.response();
+        CatbirdAction action = CatbirdAction.update(pattern, response);
+        perform(action);
+    }
+
+    public void remove(final CatbirdMock mock) {
+        RequestPattern pattern = mock.pattern();
+        CatbirdAction action = CatbirdAction.remove(pattern);
+        perform(action);
+    }
+
+    public void removeAll() {
+        perform(CatbirdAction.removeAll());
+    }
+
+    public void perform(CatbirdAction action) {
+        try {
+            send(action);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void writeMockJson(
```

Comment:
This method contains too many parameters. This is generally considered bad practice. It would be better to wrap them into [POJO](https://en.wikipedia.org/wiki/Plain_old_Java_object)